### ISSUE TITLE
RAP 4.0 RC1 - annotation @PostConstruct has no effect

### ISSUE THREAD

**Issue by clompsy** [eclipse-rap/org.eclipse.rap#226]
Updating from Eclipse RAP 3.24 to 4.0 RC1-20240828-1126 resulted for us in the methods annotated with @PostConstruct not being executed (potentially PreDestroy too).
Not sure if this is a bug or came with the Jetty upgrade, but according to the documentation, jakarta.annotation.PostConstruct and PreDestroy should work:
https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/docs/Eclipse4_RCP_Dependency_Injection.md

Are others able to use these annotations? Or did I miss something when migrating?

Easiest example is to add a PostConstruct test method on the ApplicationEntryPoint:
```
public class ApplicationEntryPoint implements EntryPoint {
	public ApplicationEntryPoint() {
	}

	@PostConstruct
	public void test() {
		System.out.println("Test");
	}

	@Override
	public int createUI() {
		(...)
	}
}
```
createUI() is called, the test() method isn't.

**Comment by ifurnadjiev** [comment]
To understand it correctly, it was working in RAP 3.2x but not in RAP 4?

**Comment by clompsy** [comment]
Correct, we updated the import to jakarta.annotation and upgraded to the RC1 mentioned in the initial post.
I can see that the RAP examples also use PostConstruct but I don't see why it would work there but doesn't in our case.

**Comment by ifurnadjiev** [comment]
Just checked our e4 demo and the @PostConstruct annotated methods are executed there.

**Comment by clompsy** [comment]
Okay, thank you, I will have to dig deeper then, not sure what would be missing that it's ignored despite no compile time errors.

**Comment by clompsy** [comment]
Got it, we had two different jakarta.annotation bundles included, the 3.0.0 caused the issue apparently.
Sorry for bothering you :)
